docs/conf.py

```diff
@@ -303,102 +366,7 @@
-            # We definitely need the preprocessor for this project.
-            ENABLE_PREPROCESSING   = YES
-            MACRO_EXPANSION        = YES
-            EXPAND_ONLY_PREDEF     = NO
-            SKIP_FUNCTION_MACROS   = NO
-            PREDEFINED             = NAMESPACE_BEGIN(nanogui)="namespace nanogui {"
-            PREDEFINED            += NAMESPACE_END(nanogui)="}"
-            PREDEFINED            += NAMESPACE_BEGIN(detail)="namespace detail {"
-            PREDEFINED            += NAMESPACE_END(detail)="}"
-            PREDEFINED            += DOXYGEN_SHOULD_SKIP_THIS
-            PREDEFINED            += DOXYGEN_DOCUMENTATION_BUILD
-            PREDEFINED            += NANOGUI_EXPORT
-        ''' % stripPath
-        # In python 3 strings and bytes are no longer interchangeable
-        if sys.version[0] == "3":
-            doxygen_input = bytes(doxygen_input, 'ASCII')
-        doxygen_proc.communicate(input=doxygen_input)
-        doxygen_proc.stdin.close()
-        if doxygen_proc.wait() != 0:
-            raise RuntimeError("Non-zero return code from 'doxygen'...")
-    except Exception as e:
-        raise Exception("Unable to execute 'doxygen': {}".format(e))
-
-
-# setup is called auto-magically for you by Sphinx
+# auto-magically called by sphinx-build
 def setup(app):
-    stripPath = ".."
-    generateDoxygenXML(stripPath)
-
-    # generate description text for the library api
-    libraryDescription = textwrap.dedent('''
-    Welcome to the developer reference to NanoGUI.  The documentation is actively being
-    developed / updated.  If you would like to help document any part of the project
-    you may be familiar with, please refer to the :ref:`developer_contribution` page.
-
-    .. note::
-       Presented below is only the C++ API. If you are using the Python API, the
-       contents below are still applicable for understanding what methods are available.
-       While the documentation for the C++ API is useful as a reference for
-       understanding what a given class does, the Python API does differ.  Please refer
-       to the more concise :ref:`nanogui_example_2` for comparing the differences
-       between the C++ and Python interfaces.
-    ''')
-
-
-    # create the dictionary to send to exhale
-    exhaleArgs = {
-        "doxygenIndexXMLPath"        : "./doxyoutput/xml/index.xml",
-        "containmentFolder"          : "./generated_api",
-        "rootFileName"               : "library_root.rst",
-        "rootFileTitle"              : "Library API",
-        "fullToctreeMaxDepth"        : 1,
-        "createTreeView"             : True,
-        "afterTitleDescription"      : libraryDescription,
-        "doxygenStripFromPath"       : ".."
-    }
-
-    # import the exhale module from the current directory and generate the api
-    from exhale import generate
-    generate(exhaleArgs)
+    # this looks in html_static_path (_static directory as set above)
+    app.add_stylesheet("entypo.css")
```

Comment:
Note: `fontcustom` generated `entypo.css` will **NOT** conflict with sphinx theme because all CSS targets are `entypo-icon-X` rather than just `icon-X`.  AKA even though this is linked to on every page, it doesn't cause issues.

Reply:
Having the icons in the HTML docs is super-nifty. I always dreaded finding the right symbols with boring recompilation cycles.

Reply:
LOL yeah it was a little tedious to get started, but the CSS generated can be parsed with regular expressions so once that was done it was good